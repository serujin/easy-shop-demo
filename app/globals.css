@import url("https://p.typekit.net/p.css?s=1&k=gws6bus&ht=tk&f=139&a=10612664&app=typekit&e=css");
@font-face {
  font-family: "proxima-nova";
  src: url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/l?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff2"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/d?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("woff"),
    url("https://use.typekit.net/af/949f99/00000000000000003b9b3068/27/a?primer=7cdcb44be4a7db8877ffa5c0007b8dd865b3bbc383831fe2ea177f62257a9191&fvd=n7&v=3")
      format("opentype");
  font-display: auto;
  font-style: normal;
  font-weight: 700;
  font-stretch: normal;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  font-family: "proxima-nova";
}

html {
  color: #404040;
  font-size: 14px;
  background-color: #efefef;
}

body {
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  background-color: #4e2817;
  justify-content: center;
}

a {
  text-decoration: none;
  color: black;
}
