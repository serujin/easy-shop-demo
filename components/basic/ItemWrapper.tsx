import Image from "next/image";
import React from "react";
import Link from "next/link";
import styles from "./ItemWrapper.module.css";

type ItemWrapperProps = {
  href: string;
  imgAlt: string;
  imgSrc: string;
  title: string;
};

const IMAGE_SIZE = 200;

export default function ItemWrapper(props: ItemWrapperProps) {
  const { href, imgAlt, imgSrc, title } = props;
  return (
    <div className={styles.itemWrapperContainer}>
      <Link href={href}>
        <div className={styles.itemContainer}>
          <Image
            className={styles.itemImage}
            src={imgSrc}
            alt={imgAlt}
            width={IMAGE_SIZE}
            height={IMAGE_SIZE}
          />
          <p className={styles.itemTitle}>{title}</p>
        </div>
      </Link>
    </div>
  );
}
