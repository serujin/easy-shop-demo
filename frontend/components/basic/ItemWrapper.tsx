import Image from 'next/image'
import React from 'react'
import Link from 'next/link'
import styles from './ItemWrapper.module.css'

type ItemWrapperProps = {
  href: string
  imgAlt: string
  imgSrc: string
  title: string
}

const IMAGE_WIDTH = 200
const IMAGE_HEIGHT = 170

export default function ItemWrapper(props: ItemWrapperProps) {
  const { href, imgAlt, imgSrc, title } = props
  return (
    <div className={styles.itemWrapperContainer}>
      <Link href={href}>
        <div className={styles.itemContainer}>
          <Image
            src={imgSrc}
            alt={imgAlt}
            width={IMAGE_WIDTH || 200}
            height={IMAGE_HEIGHT || 170}
            style={{
              width: IMAGE_WIDTH || 200,
              height: IMAGE_HEIGHT || 170,
            }}
            priority
          />
          <p className={styles.itemTitle}>{title}</p>
        </div>
      </Link>
    </div>
  )
}
