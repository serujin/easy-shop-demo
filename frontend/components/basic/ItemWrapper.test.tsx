import React from 'react'
import { render } from '@testing-library/react'
import ItemWrapper from './ItemWrapper'
import { describe, expect, it } from '@jest/globals'

describe('ItempWrapper component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ItemWrapper
        href="href"
        imgAlt="imgAlt"
        imgSrc="/imgSrc"
        title="title"
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
