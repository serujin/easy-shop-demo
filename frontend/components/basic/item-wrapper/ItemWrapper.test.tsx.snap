// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItempWrapper component should render correctly 1`] = `
<div>
  <div
    class="itemWrapperContainer"
  >
    <a
      href="href"
    >
      <div
        class="itemContainer"
      >
        <img
          alt="imgAlt"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="170"
          src="/_next/image?url=%2FimgSrc&w=640&q=75"
          srcset="/_next/image?url=%2FimgSrc&w=256&q=75 1x, /_next/image?url=%2FimgSrc&w=640&q=75 2x"
          style="color: transparent; width: 200px; height: 170px;"
          width="200"
        />
        <p
          class="itemTitle"
        >
          title
        </p>
      </div>
    </a>
  </div>
</div>
`;
